### ISSUE TITLE
Right-click fails with kalu-kde

### ISSUE THREAD

**Issue by protist** [jjk-jacky/kalu#49]
Hello, I just upgraded from KDE Plasma 5.8.5-1 -> 5.9.0-1. However it seems like [kalu-kde](https://aur.archlinux.org/packages/kalu-kde/) has some issues. (This is kalu compiled with `--enable-status-notifier` and depends on `statusnotifier`.) I can't right-click on the icon to bring up the menu. Apart from that, everything else works as expected. It hides, I can left-click and middle-click.

I'm not sure if it's important, but I get the following message when I attempt to right-click on the icon.

    (kalu:24762): GLib-GObject-CRITICAL **: g_object_set_data: assertion 'G_IS_OBJECT (object)' failed

As always in Plasma, there are a ton of other error messages at other times, e.g.

    (kalu:24762): Gtk-CRITICAL **: gtk_status_icon_set_from_icon_name: assertion 'GTK_IS_STATUS_ICON (status_icon)' failed

I also tried installing vanilla kalu, but no clicks registered with this at all (left, middle, right all failed).

**Comment by jjk-jacky** [comment]
Well, if it used to work and doesn't anymore after updating KDE Plasma,
I'd say the update in question probably broke something (as obviously
nothing changed in kalu meanwhile).

Besides, when using the status notifier interface, kalu doesn't show
nor manage its icon, and so any action (e.g. clicks) that occur on it,
that's all up to the status notifier host (i.e. some part/app of KDE)
which then triggers method on the item (i.e. kalu). So if some clicks
work but others don't, it feels like a bug on their side, not
triggering the (right) method anymore?

What you can try is running `kalu -d` from a terminal and see if you
get any clue as to what happens. I don't have access to any KDE myself,
but since nothing changed in kalu I'm not sure what could be causing
this.

Another idea: after you started kalu, try getting the menu via its
FIFO, e.g. this from a terminal:
`echo popup-menu > $XDG_RUNTIME_DIR/kalu_fifo_*`

I assume it will work, and means kalu is running properly and will popup
its menu when asked, in which case it would really indicate that the
status notifier host doesn't trigger the ContextMenu method or
something.

HTH,

**Comment by protist** [comment]
Thank you for the quick response. I tried `kalu -d`, but didn't see anything obvious. (I can post the full log if below doesn't help.)

I'm not sure if it's informative, but I tested the other third-party icons that were present in the tray, and they all work fine with right click. This includes ownCloud, Pidgin, GoldenDict, emojione-picker, and even Steam via wine.

I tested the FIFO as you suggested, 

    echo popup-menu > $XDG_RUNTIME_DIR/kalu_fifo_*
    ^Czsh: file exists: /run/user/1000/kalu_fifo_24762

It froze and I had to interrupt. There are two files here. The first works and the second fails.

    $ ls $XDG_RUNTIME_DIR/kalu_fifo_*
    /run/user/1000/kalu_fifo_12233  /run/user/1000/kalu_fifo_24762
    $ echo popup-menu > /run/user/1000/kalu_fifo_12233 # works; popup menu appears.
    $ echo popup-menu > /run/user/1000/kalu_fifo_24762
    ^Czsh: file exists: /run/user/1000/kalu_fifo_24762

Cheers.

**Comment by jjk-jacky** [comment]
If you have two fifos there, it means one is for the running kalu and
the other probably a leftover from a previous one (that likely didn't
end properly, as it should remove it upon exiting). (You can check
easily, the number at the end is the PID of kalu, so your running
kalu should be PID 12233 whilst there's no PID 24762, or it's been
re-used since but it isn't kalu)

So you can safely remove /run/user/1000/kalu_fifo_24762

As indicated though, since it works using the FIFO command, it means
kalu is working fine & will popup the menu when asked, so if
right-clicking doesn't trigger it then somehow the status notifier host
isn't triggering the ContextMenu method I'd guess, though I'm not sure
why.

But this sounds like a change in KDE, since kalu obviously hasn't
changed.

**Comment by protist** [comment]
Thanks for that. I'll post up on the KDE boards and see what I can find. Cheers.

Having said that, I guess it's something specific in the way that kalu interfaces with Plasma, because it's the only program that is exhibiting this behaviour. I'll just mention this in the KDE boards.

**Comment by jjk-jacky** [comment]
Sure thing.

Maybe ask what changed re: status notifier items between the two
versions of your upgrade, that may point to something. Let me know what
you find.

Cheers,

**Comment by protist** [comment]
I'm still waiting for an official comment in the forum, so I started filing a bug instead. However, as I was doing this, I found this similar [bug report](https://bugs.kde.org/show_bug.cgi?id=362556). It's pretty old, so it might not be related, but I just wanted to check that kalu already uses the "[updated SNI specification](https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/StatusNotifierItem/)".
